AndroidTextExtensions/common/build.gradle

```diff
@@ -13,8 +13,12 @@
 apply plugin: 'java'
 
 compileJava {
-    sourceCompatibility = 1.7
-    targetCompatibility = 1.7
+    sourceCompatibility = JavaVersion.VERSION_1_6
+    targetCompatibility = JavaVersion.VERSION_1_6
+
+    //enable compilation in a separate daemon process
+    options.fork = true
```

Comment:
This is a nice Gradle feature to allow some module to be compiled with a different JDK than the main project.
Need to have the JAVA6_HOME specified either in the global gradle.properties or the local one inside the project.